src/Feldspar/Onnx/onnxToFeld.hs

```diff
@@ -29,67 +29,111 @@
+                 , hSetBuffering, hSetBinaryMode)
 
 main :: IO ()
 main = do args <- getArgs
           let [modelFileName] = take 1 args -- First argument is file name
-          -- putStrLn modelFileName
+              modelBaseName = takeBaseName modelFileName
+              dataFileName = modelBaseName <.> "data"
+              progFileName = modelBaseName <.> "hs"
           modelBinary <- L.readFile modelFileName
-          -- putStrLn "Read it!"
           let model = either (\s -> error $ "Error: " ++ s) fst $ messageGet modelBinary
-              gr = fromMaybe (error "No graph in model") $ graph model
-          -- putStrLn $ maybe (error "Graph has no name") toStr $ G.name gr
-          let graphNodes = D.toList $ node gr
-          -- putStrLn $ show $ length graphNodes
-          -- putStrLn "module Inits where"
-          -- putStrLn "import Feldspar"
-          -- putStr   "-- Num graph initializers "
-          -- putStrLn $ show $ length $ D.toList $ G.initializer gr
-          -- putStrLn "-- Graph initializers"
-          -- putStrLn "typedef float Float;"
-          -- putStrLn $ unlines $ map showGIc $ take 1000 $ D.toList $ G.initializer gr
-          putStrLn "Graph inputs"
-          putStrLn $ unlines $ map showVI $ D.toList $ G.input gr
-          putStrLn "Graph nodes"
-          putStr $ unlines $ map showNode graphNodes
-
-showNode :: NodeProto -> String
-showNode n = "  " ++ outs ++ " = " ++ op ++ " " ++ ins ++ "\n" ++ attrs
-  where ins = unwords $ map toStr $ D.toList $ N.input n
-        -- nameStr = maybe (error "No name string") toStr $ N.name n
-        op = "onnx" ++ toStr (fromJust $ N.op_type n)
-        outs = unwords $ map toStr $ D.toList $ N.output n
-        attrs = unlines $ map showAttribute $ D.toList $ N.attribute n
-
-showAttribute :: AttributeProto -> String
-showAttribute a = "  -- " ++ nStr ++ " " ++ val nStr
+              gr = fromMaybe (error "No graph in model") $ O.graph model
+
+          -- Write the weights
+          dfile <- openFile dataFileName WriteMode
+          hSetBinaryMode dfile True
+          hSetBuffering dfile $ BlockBuffering Nothing
+          B.hPutBuilder dfile $ D.foldMap buildInitTensor $ G.initializer gr
+          hClose dfile
+
+          -- Write the program
+          pfile <- openFile progFileName WriteMode
+          hPutStr pfile $ mkProgramFile gr
+          hClose pfile
+
+-- | Extract initialized tensors
+buildInitTensor :: TP.TensorProto -> B.Builder
+buildInitTensor t = B.stringUtf8 (unwords $ showElemT dt : show (length ds) : map show ds)
+                    <> B.charUtf8 '\n'
+                    <> buildValues dt t
+  where ds = D.toList $ TP.dims t
+        dt = int2elemT $ fromJust $ TP.data_type t
+
+
+-- | Print the elements of a tensor
+buildValues :: TD.DataType -> TP.TensorProto -> B.Builder
+buildValues TD.FLOAT t = D.foldMap (\ x -> B.floatDec x <> B.char8 '\n') $ TP.float_data t
```

Comment:
Feldspar works on Windows so I think we want `B.char8 '\n'` to be `B.string8 "\n"`, same for other newlines in this file.

Reply:
Ok!